code/implementation/docker/compute/src/main/java/io/cattle/platform/docker/process/instancehostmap/DockerPostInstanceHostMapActivate.java

```diff
@@ -99,21 +108,31 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
 
         processVolumes(instance, host, state);
 
+        nativeDockerBackPopulate(instance);
+
         return null;
     }
 
-    @SuppressWarnings({ "unchecked", "rawtypes" })
+    @SuppressWarnings({ "unchecked" })
+    protected void nativeDockerBackPopulate(Instance instance) {
+        Map<String, Object> inspect = (Map<String, Object>)instance.getData().get(FIELD_DOCKER_INSPECT);
+        if (inspect == null || instance.getNativeContainer() == null || !instance.getNativeContainer().booleanValue()) {
+            return;
+        }
+        transformer.transform(inspect, instance);
```

Comment:
@alena1108 That's a good point to bring up, but I don't think these need to be wrapped up in a single transaction. Backfilling the container fields (or failing to) doesn't impact creating volumes and vice versa. I got @ibuildthecloud's opinion on it as well, to make sure I wasn't missing the mark.